app/jobs/restic_job.rb

```diff
@@ -3,6 +3,11 @@ class ResticJob < ApplicationJob
 
   def perform(*args)
     shared = "/home/deploy/lobsters/shared"
+    # Check if the shared directory exists
+    unless File.directory?(shared)
+      Rails.logger.warn "ResticJob: Shared path '#{shared}' does not exist. Skipping backup."
+      return
+    end
```

Comment:
BTW, when adding this check logic, i found the `system` call below won't raise exception even the path or restic doesn't exist on server. So I hope as Lobsters' sysadmin @pushcx  you have other ways to monitor the backup is actually working. Or maybe add `exception: true` to `system` to let it raise errors when exception happens, at least failed jobs can be found at `/jobs` 

One more thing, I'm curious how many finished jobs Lobsters currently keep given the config below, I ran an instance on my machine for 8 days, it has around 3.1k jobs in history now.

https://github.com/lobsters/lobsters/blob/c2b7f0503d1421d0389777f83fce161d0a4b609c/config/environments/production.rb#L89